src/components/SchedulePanel/components/OfficeHoursDialog/index.js

```diff
@@ -0,0 +1,526 @@
+import React, { Fragment } from 'react';
+
+import Typography from '@material-ui/core/Typography';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import FormGroup from '@material-ui/core/FormGroup';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import { gordonColors } from '../../../../theme';
+import Checkbox from '@material-ui/core/Checkbox';
+import myschedule from './../../../../services/myschedule'
+
+import './officehoursdialog.css';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+
+export default class HoursDialog extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      startHourInput: '08:00',
+      endHourInput: '17:00',
+      officeHoursOpen: false,
+      checkedC: false,
+      checkedDayofWeek: {
+        checkedMo: false,
+        checkedTu: false,
+        checkedWe: false,
+        checkedTh: false,
+        checkedFr: false,
+        checkedSa: false,
+        checkedSu: false,
+      },
+      selectedEvent: null,
+      descriptionInput: '',
+      locationInput: '',
+      descriptionValid: false,
+      locationValid: false,
+      startHourValid: true,
+      endHourValid: true,
+      formValid: false,
+      checkedValid: false,
+      formErrors: {
+        descriptionInput: '',
+        locationInput: '',
+        startHourValid: '',
+        endHourValid: '',
+      },
+    };
+    this.handleOfficeHoursClose = this.handleOfficeHoursClose.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleReset = this.handleReset.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Case : Edit(doubleclick)
+    if (this.state.selectedEvent !== nextProps.selectedEvent && nextProps.selectedEvent) {
+      this.setState({ selectedEvent: nextProps.selectedEvent }, () => {
```

Comment:
Why are we executing all of the following code within the setState? Can't we close this setState here and then execute the rest of the If block?